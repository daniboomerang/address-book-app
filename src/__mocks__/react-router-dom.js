import React from 'react';

const rrd = require('react-router-dom');

// Just render a fragment with its children
rrd.BrowserRouter = ({ children }) => <>{children}</>;
module.exports = rrd;
