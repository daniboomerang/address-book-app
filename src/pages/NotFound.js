import React from 'react';
import { Link } from 'react-router-dom';

/**
 * Adreess book not found page
 * It renders the 404 not found error message.
 */
const NotFound = () => (
  <div className="flex justify-center min-h-screen">
    <div className="w-full md:w-1/2 bg-white flex items-center justify-center">
      <div className="max-w-sm m-8">
        <div className="text-black text-5xl md:text-15xl font-black">404</div>
        <div className="w-16 h-1 bg-purple-light my-3 md:my-6" />
        <p className="text-grey-darker text-2xl md:text-3xl font-light mb-8 leading-normal">
          Sorry, the page you are looking for could not be found.
        </p>
        <Link to="/">Go home page</Link>
      </div>
    </div>
  </div>
);

export default NotFound;
