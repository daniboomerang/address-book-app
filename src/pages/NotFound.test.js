import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import NotFound from './NotFound';

describe('Not found page component', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <NotFound />
      </MemoryRouter>
    );

    expect(wrapper.text()).toContain('Sorry, the page you are looking for could not be found.');
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
