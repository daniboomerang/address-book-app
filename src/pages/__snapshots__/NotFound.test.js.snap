// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found page component should render correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotFound>
      <div
        className="flex justify-center min-h-screen"
      >
        <div
          className="w-full md:w-1/2 bg-white flex items-center justify-center"
        >
          <div
            className="max-w-sm m-8"
          >
            <div
              className="text-black text-5xl md:text-15xl font-black"
            >
              404
            </div>
            <div
              className="w-16 h-1 bg-purple-light my-3 md:my-6"
            />
            <p
              className="text-grey-darker text-2xl md:text-3xl font-light mb-8 leading-normal"
            >
              Sorry, the page you are looking for could not be found.
            </p>
            <Link
              to="/"
            >
              <LinkAnchor
                href="/"
                navigate={[Function]}
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Go home page
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </div>
    </NotFound>
  </Router>
</MemoryRouter>
`;
